module_3/src/service/User.js

```diff
@@ -0,0 +1,63 @@
+import API from './Api.js';
+import { userModel } from '../model/user.js';
+import { v4 as uuidv4 } from 'uuid';
+import statusCodes from '../../statusCodes.js';
+
+class User extends API {
+  constructor() {
+    super();
+    this.users = [];
+
+  }
+
+  findUser(id) {
+    return this.users.find(user => user.id === id);
+  }
+
+  updateUsers(updatedUser) {
+    this.users = this.users.map(user => (user.id === updatedUser.id ? updatedUser : user));
+
+    return this.users;
+  }
+
+  checkIsUserExist(id) {
+    return !!this.findUser(id);
+  }
+
+  getUser(req, res) {
+    res.status(statusCodes.ok).json(this.findUser(req.params.id));
+  }
+
+  addUser(req, res) {
+    const user = super.filterModel(userModel, req.body);
+    const id = uuidv4();
+    this.users.push({ id, ...user, isDeleted: false });
+    res.status(statusCodes.ok).json(id);
+  }
```

Comment:
UserService must not be coupled with `req` or `res` objects. It must be responsible only for managing user data.